core-rules/src/test/java/org/hisp/dhis/rules/functions/RuleFunctionAddControlDigitsShould.java

```diff
@@ -0,0 +1,75 @@
+package org.hisp.dhis.rules.functions;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+import static org.assertj.core.api.Java6Assertions.fail;
+
+import org.hisp.dhis.rules.RuleVariableValue;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+
+@RunWith(JUnit4.class)
+public class RuleFunctionAddControlDigitsShould {
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void evaluate_correct_add_control_digits_value() {
```

Comment:
The tests should serve as documentation but I don't undertand the behavior with current tests, Does it exist documentation about this function in dhis 2 docs? I have not found it [here](https://docs.dhis2.org/2.25/en/user/html/manage_program_rule.html) This function is not enable via tracker capture web, only exists in dhis2.angular.services.js. We can not know if javascript implementation is valid because don't exists documentation about it and community is not using this function via web. Has sense that we adding this function to SDK?